include/painless/parameter.h

```diff
@@ -79,7 +83,24 @@ class Parameter {
         m_default_value(default_value),
         m_current_value(default_value),
         m_file_watcher{} {
-    mkdir(BASE_PATH, 0777);
+    m_base_path = get_base_path();
+
+    if (mkdir(m_base_path.c_str(), 0777) != 0) {
+      if (errno == EEXIST) {
+        struct stat sb;
+        if (lstat(m_base_path.c_str(), &sb) != 0) {
+          throw std::runtime_error{"cannot stat() " + m_base_path +
+                                   std::strerror(errno)};
+        }
+        if ((sb.st_mode & S_IFMT) != S_IFDIR) {
+          throw std::runtime_error{"mkdir " + m_base_path +
+                                   " failed: " + std::strerror(errno)};
+        }
+      } else {
+        throw std::runtime_error{"mkdir " + m_base_path +
+                                 " failed: " + std::strerror(errno)};
+      }
+    }
```

Comment:
If `m_base_path` already exists we have to check whether it is a directory (in which case everything is fine).  Otherwise this is an irrecoverable error.

Reply:
Thank you for writing this code. Earlier, I went through the `mkdir` manpage and saw the list of possible error codes, and decided to postpone the error handling :smile: (note that errors will also be captured by the file creation below)